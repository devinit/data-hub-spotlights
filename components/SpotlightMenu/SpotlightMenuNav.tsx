import classNames from 'classnames';
import React, { Children, FunctionComponent, isValidElement, cloneElement } from 'react';

interface ComponentProps {
  active?: boolean;
  onClick?: () => void;
  caption: string;
  onShowAll?: () => void;
}

const SpotlightMenuNav: FunctionComponent<ComponentProps> = ({ active, caption, children, onClick, onShowAll }) => {
  return (
    <nav className={classNames('countries-menu-list animated', { inactive: !active })}>
      <a
        className="countries-menu-list__item countries-menu-list__parent countries-menu-list__item--open js-countries-menu-trigger"
        onClick={onClick}
      >
        {caption}
      </a>
      <a
        className={classNames('countries-menu__profile countries-menu__link js-profile-item', { hide: !onShowAll })}
        aria-hidden="true"
        title={`View ${caption}`}
        onClick={onShowAll}
        data-testid="spotlight-nav-view"
      >
        View
      </a>
      <style jsx>{`
        cursor: pointer;
      `}</style>
      {Children.map(children, (child) => isValidElement(child) && cloneElement(child, { active }))}
    </nav>
  );
};

export default SpotlightMenuNav;
