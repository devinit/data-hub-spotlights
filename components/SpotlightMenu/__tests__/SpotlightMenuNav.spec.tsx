/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import SpotlightMenuNav from '../SpotlightMenuNav';

describe('SpotlightMenuNav', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SpotlightMenuNav caption="Uganda" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenuNav caption="Uganda">
        <div>Testing</div>
      </SpotlightMenuNav>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
