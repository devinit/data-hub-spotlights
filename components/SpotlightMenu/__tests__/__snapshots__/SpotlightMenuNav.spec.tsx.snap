// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightMenuNav renders children correctly 1`] = `
<nav
  className="countries-menu-list animated inactive"
>
  <a
    className="countries-menu-list__item countries-menu-list__parent countries-menu-list__item--open js-countries-menu-trigger"
    href="#"
  >
    Uganda
  </a>
  <a
    aria-hidden="true"
    className="countries-menu__profile countries-menu__link js-profile-item"
    title="View Uganda"
  >
    View
  </a>
  <style
    jsx={true}
  >
    
        cursor: pointer;
      
  </style>
  <div>
    Testing
  </div>
</nav>
`;

exports[`SpotlightMenuNav renders correctly 1`] = `
<nav
  className="countries-menu-list animated inactive"
>
  <a
    className="countries-menu-list__item countries-menu-list__parent countries-menu-list__item--open js-countries-menu-trigger"
    href="#"
  >
    Uganda
  </a>
  <a
    aria-hidden="true"
    className="countries-menu__profile countries-menu__link js-profile-item"
    title="View Uganda"
  >
    View
  </a>
  <style
    jsx={true}
  >
    
        cursor: pointer;
      
  </style>
</nav>
`;
